ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -48,33 +49,63 @@ import kotlin.collections.component2
 import kotlin.collections.set
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.reactive.asFlow
 
 /** View model for the Data Collection fragment. */
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltViewModel
 class DataCollectionViewModel
 @Inject
 internal constructor(
   private val viewModelFactory: ViewModelFactory,
   private val submissionRepository: SubmissionRepository,
+  private val locationOfInterestRepository: LocationOfInterestRepository,
   private val locationOfInterestHelper: LocationOfInterestHelper,
   private val popups: Provider<EphemeralPopups>,
   private val navigator: Navigator,
   @ApplicationScope private val externalScope: CoroutineScope,
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
-  private val savedStateHandle: SavedStateHandle
+  private val savedStateHandle: SavedStateHandle,
+  surveyRepository: SurveyRepository,
 ) : AbstractViewModel() {
 
-  val submission: @Hot(replays = true) LiveData<Submission>
-  val jobName: @Hot(replays = true) LiveData<String>
-  val loiName: @Hot(replays = true) LiveData<String>
+  // TODO(#1541): Set loiId once Suggest LOI task is completed.
+  private val loiId: StateFlow<String?> = MutableStateFlow(savedStateHandle["locationOfInterestId"])
+  private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
+  private val job: Job =
+    activeSurvey.getJob(requireNotNull(savedStateHandle["jobId"])).orElseThrow()
+
+  val surveyId: String = surveyRepository.lastActiveSurveyId
+  val submission: StateFlow<Submission?> =
+    loiId
+      .flatMapLatest {
+        if (it == null) flowOf(null)
+        else submissionRepository.createSubmission(surveyId, it).toFlowable().asFlow()
+      }
+      .stateIn(viewModelScope, SharingStarted.Lazily, null)
+
+  val jobName: StateFlow<String> =
+    MutableStateFlow(job.name ?: "").stateIn(viewModelScope, SharingStarted.Lazily, "")
+  val loiName: StateFlow<String> =
+    loiId
+      .flatMapLatest { id ->
+        if (id == null) flowOf("")
+        else
+          locationOfInterestRepository
+            .getOfflineLocationOfInterest(surveyId, id)
+            .toFlowable()
+            .asFlow()
```

Comment:
IMO, we should consider moving this to the repository so that we can incrementally remove usages from other places as well. Same comment for the above usage of `.asFlow()` as well.